src/query/generated/proto/rpcpb/query.proto

```diff
@@ -90,3 +93,64 @@ message M3Segment {
 	int64 startTime = 3;
 	int64 blockSize = 4;
 }
+
+message SearchRequest {
+	oneof matchers {
+		TagMatchers tagMatchers = 1;
+	}
+}
+
+message M3TagProperty {
+	bytes id             = 1;
+	bytes compressedTags = 2;
+}
+
+message M3TagProperties {
+	repeated M3TagProperty properties = 1;
+}
+
+message TagProperty {
+	bytes key             = 1;
+	repeated bytes values = 2;
+}
+
+message TagProperties {
+	repeated TagProperty properties = 1;
+}
+
+message SearchResponse {
+	oneof value {
+		TagProperties decompressed = 1;
```

Comment:
Hm for this case, there should really only be one tag value per tag key, should we create two types `TagProperty` for the `SearchResponse` case and `TagValues` for the `CompleteTagsResponse` case that has multiple tag values (to match the naming of the `TagNames` type)?

Reply:
Otherwise this looks good 👍